daemon/cmd/daemon_main.go

```diff
@@ -253,9 +253,17 @@ func init() {
 	flags.StringSlice(option.DebugVerbose, []string{}, "List of enabled verbose debug groups")
 	option.BindEnv(option.DebugVerbose)
 
-	flags.StringSliceP(option.Device, "d", []string{}, "List of devices facing cluster/external network for attaching bpf_netdev (first device should be one used for direct routing if tunneling is disabled)")
+	flags.StringP(option.Device, "d", "",
+		fmt.Sprintf("Device facing cluster/external network for attaching bpf_host. Deprecated in favor of --%s", option.Devices))
+	flags.MarkDeprecated(option.Device, "This option will be removed in v1.10")
 	option.BindEnv(option.Device)
 
+	flags.StringSlice(option.Devices, []string{}, "List of devices facing cluster/external network for attaching bpf_host")
+	option.BindEnv(option.Devices)
+
+	flags.String(option.DirectRoutingDevice, "", "Device name used to connect nodes in direct routing mode (required only by BPF NodePort; derived automatically if empty)")
+	option.BindEnv(option.DirectRoutingDevice)
```

Comment:
I would briefly state from where (`derived automatically if empty`) this is derived.

Reply:
I presume this has to be part of `--devices` list. Perhaps should be stated.

Reply:
> I presume this has to be part of --devices list. Perhaps should be stated.

It's derived from a device which as a k8s node IP (the device is used by the direct routing routes). If the device was not specified within `--devices`, then it will be added to the list.

Reply:
Given we would have detection working nicely for it, what k8s use-case do you have in mind where we would still the DirectRoutingDevice specified? (I just think from a user pov, it's confusing.)

Reply:
When a user manages routing by itself (e.g. via BGP; `--auto-direct-node-routes=false`). Then they might want to control which interface should be used for direct routing.

Reply:
Fair enough, if there are known users that need this control beyond default detection, then lets add. (I just think that all these knobs to specify devices are just limited since it requires all nodes to be exact homogeneous wrt their device [and device naming] setup in order to work.)